sdk/Lusid.Sdk.Tests/Utilities/PollyApiRetryHandlerTests.cs

```diff
@@ -0,0 +1,409 @@
+            var policy2 = Policy
+                .HandleResult<IRestResponse>(apiResponse => apiResponse.StatusCode == (HttpStatusCode) statusCodeResponse2)
+                .Retry(retryCount: 3, onRetry: (result, i) => policy2TriggerCount++);
+            RetryConfiguration.RetryPolicy = policy1.Wrap(policy2);
+
+            // Calling GetPortfolio or any other API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any");
+
+            Assert.That(policy1TriggerCount, Is.EqualTo(2));
+            Assert.That(policy2TriggerCount, Is.EqualTo(1));
+            // Response needs to be a valid, deserializable json, otherwise we get null. To be fixed in another PR.
+            Assert.That(sdkResponse, Is.Null);
+        }
+
+        // Show that polly retries are triggered on regular API timeouts as well.
+        // Default timeout config is 100000 seconds (1min40s)
+        [Test]
+        public void CallGetPortfoliosApi_WhenRequestTimeExceedsTimeoutConfigured_PollyRetryIsStillTriggered()
+        {
+            var timeoutAfterMillis = GlobalConfiguration.Instance.Timeout;
+            const int returnedStatusCode = 200; // Doesn't matter what code is on timeout, will always return 0
+            const int expectedNumberOfRetries = 1;
+            // First call will cause a timeout
+            AddMockHttpResponseToQueue(_httpListener, statusCode: returnedStatusCode, responseContent: "",
+                timeoutAfterMillis + 10);
+            // No timeout on the second call
+            AddMockHttpResponseToQueue(_httpListener, statusCode: returnedStatusCode, responseContent: "");
+            var retryCount = 0;
+            RetryConfiguration.RetryPolicy =
+                Policy.Wrap(
+                    PollyApiRetryHandler.DefaultFallbackPolicy,
+                    Policy
+                        // Use default internal exception retry condition checker
+                        .HandleResult<IRestResponse>(PollyApiRetryHandler.GetInternalExceptionRetryCondition)
+                        .Retry(retryCount: expectedNumberOfRetries,
+                            onRetry: (result, i) => retryCount++)
+                );
+
+            // Calling GetPortfolio or any other API triggers the flow that triggers polly
+            var sdkResponse = _apiFactory.Api<IPortfoliosApi>().GetPortfolio("any", "any");
+
+            Assert.That(retryCount, Is.EqualTo(expectedNumberOfRetries));
+            // Todo: In the future 0 error codes with throw an error after retries exceeded
+            Assert.That(sdkResponse, Is.Null);
+        }
+
+
+        [Test]
+        public void CreateLusidFactory_WhenRetryPolicyIsNull_AssignsDefaultRetryPolicy()
+        {
+            RetryConfiguration.RetryPolicy = null;
+
+            var newFactory = new LusidApiFactory(new Configuration());
+
+            Assert.That(RetryConfiguration.RetryPolicy, Is.Not.Null);
```

Comment:
Can we assert what it should be? Can we do this for sync and async?

Reply:
Done

Reply:
Nice!